modules/registry-configuring-registry-storage-rhodf-cephrgw.adoc

```diff
@@ -0,0 +1,104 @@
+
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have access to the {product-title} web console.
+* You installed the `oc` CLI.
+* Install the {rh-storage-first} Operator to provide object storage and `cephfs` file storage.
+
+.Procedure
+
+. Create the object bucket claim using the `ocs-storagecluster-ceph-rgw` storage class. For example:
++
+[source,terminal]
+----
+cat <<EOF | oc apply -f -
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: rgwtest
+  namespace: openshift-storage
+spec:
+  storageClassName: ocs-storagecluster-ceph-rgw
+  generateBucketName: rgwtest
+EOF
+----
+
+. Get the bucket name by entering the following command:
++
+[source,terminal]
+----
+$ bucket_name=$(oc get obc rgwtest -o jsonpath='{.spec.bucketName}')
+----
+
+. Get the AWS credentials by entering the following commands:
++
+[source,terminal]
+----
+$ AWS_ACCESS_KEY_ID=$(oc get secret -n openshift-storage rgwtest -o yaml | grep -w "AWS_ACCESS_KEY_ID:" | head -n1 | awk '{print $2}' | base64 --decode)
+----
++
+[source,terminal]
+----
+$ AWS_SECRET_ACCESS_KEY=$(oc get secret -n openshift-storage rgwtest -o yaml | grep -w "AWS_SECRET_ACCESS_KEY:" | head -n1 | awk '{print $2}' | base64 --decode)
+----
+
+. Create a secret with the AWS credentials for the new bucket under `openshift-image-registry project` by entering the following command:
++
+[source,terminal]
+----
+$ oc create secret generic image-registry-private-configuration-user --from-literal=REGISTRY_STORAGE_S3_ACCESSKEY=${AWS_ACCESS_KEY_ID} --from-literal=REGISTRY_STORAGE_S3_SECRETKEY=${AWS_SECRET_ACCESS_KEY} --namespace openshift-image-registry
+----
+
+. Create a encryption route for Ceph RGW by entering the following command:
++
+[source,terminal]
+----
+$ oc create route reencrypt <route_name> --service=rook-ceph-rgw-ocs-storagecluster-cephobjectstore --port=https -n openshift-storage
+----
++
+.. Get the route host by entering the following command:
++
+[source,terminal]
+----
+$ route_host=$(oc get route <route_name> -n openshift-storage -o=jsonpath='{.spec.host}')
+----
+. Create config map that uses an ingress certificate by entering the following commands:
++
+[source,terminal]
+----
+$ oc extract secret/router-certs-default  -n openshift-ingress  --confirm
+----
++
+[source,terminal]
+----
+$ oc create configmap image-registry-s3-bundle --from-file=ca-bundle.crt=./tls.crt  -n openshift-config
+----
+
+. Configure the image registry to use the Ceph RGW object storage by entering the following command:
++
+[source,terminal]
+----
+$ oc patch config.image/cluster -p '{"spec":{"managementState":"Managed","replicas":2,"storage":{"managementState":"Unmanaged","s3":{"bucket":"${bucket_name}","region":"us-east-1","regionEndpoint":"https://${route_host}","virtualHostedStyle":false,"encrypt":true,"trustedCA":{"name":"user-ca-bundle"}}}}}' --type=merge
```

Comment:
The cmd should be, please notice the backslash
oc patch config.image/cluster -p '{"spec":{"managementState":"Managed","replicas":2,"storage":{"managementState":"Unmanaged","s3":{"bucket":'\"${bucket_name}\"',"region":"us-east-1","regionEndpoint":'\"https://${route_host}\"',"virtualHostedStyle":false,"encrypt":true,"trustedCA":{"name":"user-ca-bundle"}}}}}' --type=merge